File: typescript-vitest/src/varar/tables-and-docstrings.steps.ts
import { steps } from '@varar/varar'

const { sensor } = steps()

sensor('Uppercase each one:', (_state, rows: ReadonlyArray<ReadonlyArray<string>>) => {
  return rows.slice(1).map(([before]) => ({ before, after: (before ?? '').toUpperCase() }))
})

sensor('Greet {word}:', (_state, name, _body: string) => {
  return [name, `Hello, ${name}!\n`]
})

File: python-pytest/tests/varar/tables_and_docstrings.steps.py
from varar import steps

param, stimulus, sensor = steps()


@sensor("Uppercase each one:")
def _(state, rows):
    return [{"before": before, "after": before.upper()} for before, *_ in rows[1:]]


@sensor("Greet {word}:")
def _(state, name, doc):
    return [name, f"Hello, {name}!\n"]
